Technolab.OnlineLibrary.Web/Controllers/AccountController.cs

```diff
@@ -1,14 +1,54 @@
 ﻿using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using Technolab.OnlineLibrary.Web.Models;
+using Technolab.OnlineLibrary.Web.ViewModels;
 
 namespace Technolab.OnlineLibrary.Web.Controllers
 {
     [Authorize(Policy = AuthorizationPolicies.Users)]
     public class AccountController : Controller
     {
+        public AccountController(ILibraryDbContextFactory contextFactory)
+        {
+            this.ContextFactory = contextFactory;
+        }
+
         public ActionResult Index()
         {
             return View();
         }
+
+        [HttpGet]
+        public ActionResult ChangePassword()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public ActionResult ChangePassword(ChangePasswordViewModel model)
+        {
+            if (model.ConfirmNewPassword == model.NewPassword)
+            {
+                if (!string.IsNullOrWhiteSpace(model.NewPassword))
+                {
+                    int userId = Convert.ToInt16(HttpContext.User.GetId());
+                    using var context = ContextFactory.Create();
+                    var user = context.Users.Find(x => x.Id == userId);
+
+                    user.ResetPassword(model.NewPassword);
+                    context.SaveChanges();
+
+                    return View(new ChangePasswordViewModel { PasswordChangedSuccessfully = true });
+                }

```

Comment:
What if `model.NewPassword` is null or empty? You should return an error in this case as well.

Reply:
if `model.NewPassword` is empty I'm currently returning a "change password" view saying that the missing fields are required. Is this not better than returning an error?
![Screenshot_1](https://github.com/user-attachments/assets/47da8ee1-fd6b-4368-8a63-0541f1a1b648)